zarr/util.py

```diff
@@ -19,10 +19,24 @@
 }
 
 
+# from
+# https://stackoverflow.com/questions/50916422/python-typeerror-object-of-type-int64-is-not-json-serializable/50916741
+class NpEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.integer):
+            return int(obj)
+        elif isinstance(obj, np.floating):
+            return float(obj)
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        else:
+            return super(NpEncoder, self).default(obj)
+
+
 def json_dumps(o):
     """Write JSON in a consistent, human-readable way."""
     return json.dumps(o, indent=4, sort_keys=True, ensure_ascii=True,
-                      separators=(',', ': ')).encode('ascii')
+                      separators=(',', ': '), cls=NpEncoder).encode('ascii')
```

Comment:
Wouldn't we need a way to decode NumPy values too? How would we do that?

Reply:
I don't understand this. 

What do you mean with numpy values? The encoder is just what it does. It adds encoders for three numpy data types: integers, floats and arrays.

Reply:
I think @jakirkham means, when reading the JSON back, do you want to convert the array back into a numpy array, or are you happy to see it as a bare list? 

I.e., this encoder will not round-trip, so some users might get a surprise when the store an array and then read back and get a list.

Reply:
I think it's definitely OK to return Python builtin scalars instead of NumPy scalars. Half the time users don't even realize they're using NumPy scalars because NumPy uses the same repr 😒 

I'm less sure about whether or not it's a good idea to automatically convert arrays into lists.